app/controllers/incomeSources/add/BusinessReportingMethodController.scala

```diff
@@ -0,0 +1,198 @@
+    val postAction: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessReportingMethodController.submitAgent(id) else
+      controllers.incomeSources.add.routes.BusinessReportingMethodController.submit(id)
+    val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
+    val redirectUrl: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessAddedController.showAgent(id) else
+      controllers.incomeSources.add.routes.BusinessAddedController.show(id)
+
+    if (incomeSourcesEnabled) {
+      businessReportingMethodService.checkITSAStatusCurrentYear.flatMap {
+        case true =>
+          businessReportingMethodService.getBusinessReportingMethodDetails(id).map {
+            case Some(viewModel) =>
+              Ok(view(
+                addBusinessReportingMethodForm = AddBusinessReportingMethodForm.form,
+                businessReportingViewModel = viewModel,
+                postAction = postAction,
+                isAgent = isAgent)(user, messages))
+            case None =>
+              Redirect(redirectUrl)
+          }.recover {
+            case err: Exception =>
+              Logger("application").error(s"${if (isAgent) "[Agent]"}" + s"Error getting BusinessReportingMethodDetails : $err")
+              errorHandler.showInternalServerError()
+          }
+
+        case false => Future.successful(Redirect(redirectUrl))
+
+      }.recover {
+        case err: InternalServerException =>
+          Logger("application").error(s"${if (isAgent) "[Agent]"}" + s"Error getting ITSA Status : $err")
+          errorHandler.showInternalServerError()
+      }
+
+
+    } else {
+      Future.successful(Ok(customNotFoundErrorView()(user, messages)))
+    } recover {
+      case ex: Exception =>
+        Logger("application").error(s"${if (isAgent) "[Agent]"}" +
+          s"Error getting BusinessReportingMethodController page: ${ex.getMessage}")
+        errorHandler.showInternalServerError()
+    }
+  }
+
+  private def handleSubmitRequest(isAgent: Boolean, id: String)(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Result] = {
+    val incomeSourcesEnabled: Boolean = isEnabled(IncomeSources)
+    val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
+    val redirectUrl: Call = if (isAgent) routes.BusinessAddedController.showAgent(id) else routes.BusinessAddedController.show(id)
+    val submitUrl: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessReportingMethodController.submitAgent(id) else
+      controllers.incomeSources.add.routes.BusinessReportingMethodController.submit(id)
+
+    if (incomeSourcesEnabled) {
+      AddBusinessReportingMethodForm.form.bindFromRequest().fold(
+        hasErrors => {
+          val updatedForm = AddBusinessReportingMethodForm.updateErrorMessagesWithValues(hasErrors)
+
+          if (updatedForm.hasErrors) {
```

Comment:
does the hasErrors function not already predicate that there are errors. In which case, is this check redundant?

Reply:
this is to check false errors with form when the taxYear 1 is crystallised, where we don't show taxYear 1 reporting method radio option.  It evaluates as an error